url1a745d67fc_sobercheg_InterviewTraining_tgz-pJ8-leetcode_oj_Solution2TestJ8/src/crackinginterview/chap2linkedlists/KthToLastInList.java

```diff
@@ -0,0 +1,44 @@
+package crackinginterview.chap2linkedlists;
+
+import common.ListNode;
+
+/**
+ * Created by Sobercheg on 12/6/13.
+ */
+public class KthToLastInList {
+
+    static class MutableInt {
+        int value = 0;
+
+        MutableInt(int value) {
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+            return "" + value;
+        }
+    }
+
+    public ListNode getKthToLastInList(ListNode head, int k, MutableInt index) {
+        if (head == null) return null;
+        ListNode node = getKthToLastInList(head.next, k, index);
+        // but this is wrong!
+        index.value = index.value + 1;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;index&#39; (index: 2 | type: MutableInt). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;k&#39;, &#39;head&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).